components/proxy/plugins/jsonselect/plugin.go

```diff
@@ -104,6 +104,10 @@ func (e JSONSelectEncoder) EncodeEntry(entry zapcore.Entry, fields []zapcore.Fie
 	// TODO(leo): switch back to EachKey (see git history) for perf reasons when fixed
 	for idx, paths := range e.getters {
 		val, typ, _, err := jsonparser.Get(buf.Bytes(), paths...)
+		if err == jsonparser.KeyPathNotFoundError {
```

Comment:
We need this check to handle access to headers that are not available in all requests, e.g.:

```
{cacheStatus:resp_headers>X-Cache-Status>[0]}
```

Without this check, this would lead to an error when the header `X-Cache-Status` is missing.

/cc @leodido